docs/site/clojure-lsp.md

```diff
@@ -8,24 +8,42 @@ description: Learn about how Calva leverages clojure-lsp for a lot of the featur
+    Calva defaults to using the `latest` clojure-lsp released. To use a different version of clojure-lsp, see the [configuration](#configuration) section. **Calva does not use the clojure-lsp installed on your system, unless you [set the path for clojure-lsp](#using-a-custom-clojure-lsp-native-binary) to the installed binary in your settings**. You can see what version is being used by running the `Clojure-lsp Server Info` command, which will also show the version of clj-kondo that's being used as well as other info.
 
-## The LSP server starts automatically
+## The LSP server lifecycle
 
-Unless you set `calva.enableClojureLspOnStart` to `false`, Calva will automatically start clojure-lsp. And you won't need to install antyting, because Calva handles that. It can take a while for clojure-lsp to start, especially the first time for a new project, when clojure-lsp (via `clj-kondo`) indexes the project files.
+By default you won't need to install/setup anything as Calva handles that for you by automatically downloading the latest clojure-lsp binary. It can take a while for clojure-lsp to start, especially the first time opening a new project, as clojure-lsp (via `clj-kondo`) indexes the project files.
 
-## LSP server process control
+Calva is able to automatically start the clojure-lsp server for you and can be configured to start the server under various different conditions. These behaviours can be configured through the `calva.enableClojureLspOnStart` setting, which takes the following options:
 
-It is highly recommended to let clojure-lsp start automatically, and most often you can just let it do its wonderful magic without bothering with it. There are times when you might want to control and inspect the process, though. Calva has commands for:
++ "when-workspace-opened"
++ "when-file-opened"
++ "never"
+
+#### "when-workspace-opened" [default]
+
+When set to `"when-workspace-opened"` Calva will start the clojure-lsp in the root of all opened vscode workspaces. All clojure files in a workspace will be services by the lsp server running in that workspace. This behaviour requires that you are opening workspaces with a valid clojure project in the root (the directory must contain a `deps.edn`, `project.clj` or `shadow-cljs.edn` file).
+
+This is the default auto-start behaviour.
+
+#### "when-file-opened"
+
+When set to `"when-file-opened"` Calva will attempt to start the clojure-lsp whenever a clojure file is opened. The LSP server will be started in the outermost valid Clojure project or will fall back to starting in the workspace root if no valid Clojure project can be found. A directory is considered a Clojure project if it contains typical Clojure project files such as a `deps.edn`, `project.clj` or `shadow-cljs.edn` file. When working in a mono-repo style project or in a multi-workspace vscode configuration you may have multiple LSP servers running for each independent Clojure project root.
```

Comment:
```suggestion
When set to `"when-file-opened"` Calva will attempt to start the clojure-lsp server whenever a Clojure file is opened. The LSP server will be started in the outermost valid Clojure project or will fall back to starting in the workspace root if no valid Clojure project can be found. A directory is considered a Clojure project if it contains typical Clojure project files such as a `deps.edn`, `project.clj`, or `shadow-cljs.edn` file. When working in a mono-repo style project or in a multi-workspace VS Code configuration you may have multiple LSP servers running for each independent Clojure project root.
```

Reply:
> or will fall back to starting in the workspace root if no valid Clojure project can be found

Clojure-lsp will start even if no project is found in the workspace root? Doesn't it need at least a project file in the workspace root to start for the workspace root?

Reply:
The clojure-lsp does still work without a project-file however it is limited in what it can do, afaict.

So using that as a fallback allows things to continue to work, just slightly degraded.